File: 011_container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max=0;
        int min_height;
        auto left=height.begin(), right=height.end()-1;
        while (left<right) {
            min_height=min(*left,*right);
            if ((right-left) * min_height > max) max=(right-left)*min_height;
            
            if (*left<=*right) left++;
            else right--;
        }
       
        return max;
    }
};

File: algorithms/python3/011_container_with_most_water.py
class Solution:
    def maxArea(self, height):
        i,j=0,len(height)-1
        maxarea=(j-i)*min(height[i],height[j])
        while i<j:
            if height[i]>height[j]:
                j=j-1
            else:
                i=i+1
            maxarea=max(maxarea,(j-i)*min(height[i],height[j]))
        return maxarea


res=Solution()

print(res.maxArea([1,8,6,2,5,4,8,3,7]))

